app/models/institution_program.rb

```diff
@@ -1,4 +1,83 @@
 # frozen_string_literal: true
 
 class InstitutionProgram < ApplicationRecord
+  validates :version, :institution, :facility_code, :description, presence: true
+
+  PROGRAM_TYPES = %w[
+    IHL
+    NCD
+    OJT
+    FLGT
+    CORR
+  ].freeze
+
+  belongs_to :institution
+  delegate :dod_bah, to: :institution
+  delegate :preferred_provider, to: :institution
+
+  def institution_name
+    institution.institution
+  end
+
+  def city
+    institution.physical_city
+  end
+
+  def state
+    institution.physical_state
+  end
+
+  def country
+    institution.physical_country
+  end
+
+  def va_bah
+    institution.bah
+  end
+
+  # Given a search term representing a partial school name, returns all
+  # schools starting with the search term.
+  #
+  def self.autocomplete(search_term, limit = 6)
+    select('id, facility_code as value, description as label')
+      .where('lower(description) LIKE (?)', "#{search_term}%")
+      .limit(limit)
+  end
+
+  # Finds exact-matching facility_code or partial-matching school and city names
+  #
+  scope :search, lambda { |search_term, _include_address = false|
+    return if search_term.blank?
+
+    clause = [
+      'institution_programs.facility_code = (:facility_code)',
```

Comment:
If we are going for data normalization, we no longer need `facility_code` or `version` in the `institution_programs` table.  That data already exists in `institutions`.  I think that's all of the duplicated fields.

Reply:
That sounds good. I can make the changes in this PR to stop using `institutions_programs.facility_code` and then drop that in a following migration PR.